### PR TITLE
[PM-11270] Hide all new UI behind onboarding flow flag.

### PR DESCRIPTION
**Pull request by dseverns-livefront**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-11270

## 📔 Objective

- Hide the new changes for the onboarding flow designs behind feature flags in the screens used for new email verification too 
- this is part one, the first two screens (start and check email)

## 📸 Screenshots

https://github.com/user-attachments/assets/be8b2f23-bb74-4004-b0c9-d2a001ab910e

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailViewModel.kt [hunk 1]
```diff
@@ -17,25 +21,52 @@ private const val KEY_STATE = "state"
...
         stateFlow
             .onEach { savedStateHandle[KEY_STATE] = it }
             .launchIn(viewModelScope)
+        // Listen for changes on the onboarding feature flag.
+        featureFlagManager
+            .getFeatureFlagFlow(FlagKey.OnboardingFlow)
+            .map {
+                CheckEmailAction.Internal.OnboardingFeatureFlagUpdated(it)
+            }
+            .onEach(::handleAction)
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailViewModel.kt:44]: should be `::sendAction`

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreen.kt [hunk 2]
```diff
@@ -198,9 +209,136 @@ private fun CheckEmailContent(
     }
 }
 
+@Suppress("LongMethod")
+@Composable
+private fun CheckEmailLegacyContent(
+    email: String,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreen.kt:215]: Can I assume this was all wholesale copied from the old code?

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreen.kt [hunk 3]
```diff
@@ -198,9 +209,136 @@ private fun CheckEmailContent(
...
+                },
+            )
+        }
+    }
+}
+
 @Preview(showBackground = true)
 @Composable
-private fun CheckEmailScreenPreview() {
+private fun CheckEmailScreenNewUi_preview() {
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreen.kt:341]: 👍

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailViewModel.kt [hunk 4]
```diff
@@ -17,25 +21,52 @@ private const val KEY_STATE = "state"
  */
 @HiltViewModel
 class CheckEmailViewModel @Inject constructor(
+    featureFlagManager: FeatureFlagManager,
     savedStateHandle: SavedStateHandle,
 ) : BaseViewModel<CheckEmailState, CheckEmailEvent, CheckEmailAction>(
     initialState = savedStateHandle[KEY_STATE]
         ?: CheckEmailState(
             email = CheckEmailArgs(savedStateHandle).emailAddress,
+            showNewOnboardingUi = false,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailViewModel.kt:30]: Can this be:

```kotlin
showNewOnboardingUi = featureFlagManager.getFeatureFlag(FlagKey.OnboardingFlow)
```

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/startregistration/StartRegistrationViewModel.kt [hunk 5]
```diff
@@ -73,6 +79,14 @@ class StartRegistrationViewModel @Inject constructor(
                 )
             }
             .launchIn(viewModelScope)
+        // Listen for changes on the onboarding feature flag.
+        featureFlagManager
+            .getFeatureFlagFlow(FlagKey.OnboardingFlow)
+            .map {
+                OnboardingFeatureFlagUpdated(it)
+            }
+            .onEach(::handleAction)
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/startregistration/StartRegistrationViewModel.kt:88]: `sendAction` here too

File: app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreenTest.kt [hunk 6]
```diff
@@ -63,6 +67,12 @@ class CheckEmailScreenTest : BaseComposeTest() {
         }
     }
 
+    @Test
+    fun `login button click should send LoginTap action`() {
+        mutableEventFlow.tryEmit(CheckEmailEvent.NavigateBackToLanding)
+        TestCase.assertTrue(onNavigateToLandingCalled)
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreenTest.kt:73]: Can we use `Assert` and we also try to import the functions:

```kotlin
import org.junit.Assert.assertTrue
```

Comment (dseverns-livefront) [reply app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreenTest.kt:73]: oop

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreen.kt (hunk 2 above)
Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailScreen.kt:215]: 💯

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailViewModel.kt (hunk 4 above)
Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/checkemail/CheckEmailViewModel.kt:30]: changed in both

**Review by david-livefront — APPROVED** [review]

**Pull request merged**